### DISCUSSION TITLE
Help setting up custom surround

### DISCUSSION THREAD

**Discussion by videbar** [kylechui/nvim-surround#268] [Q&A]
Hi, I'm trying to set up a custom surround to work with wikilinks that look `[[like this]]`, i.e., simply a pair of square brackets.

I've added this to my config and the `add` part works great, but I can't get `delete` to work. I've tried different lua expressions, but when trying to delete the separators, it inserts a new `]` instead.

```lua
require("nvim-surround").setup({
    surrounds = {
        ["w"] = {
            add = { "[[", "]]" },
            find = function()
                return require("nvim-surround.config").get_selection { motion = "a]a]" }
            end,
            delete = "^(%[%[)().-(%]%])()$"

        }
    }
})
```
Any idea on what I may be doing wrong?

**Comment by kylechui** [comment]
Try changing `a]a]` -> `2a]`; I think the former is a singular motion `a]`, and then inserts a `]` after the cursor.

**Reply by videbar** [reply]
Thanks! that was indeed the problem. Now both `add` and `delete` work, but I can't get `change` to work as well.
 I've updated my config to look like this:

```lua
require("nvim-surround").setup({
    surrounds = {
        -- Wikilinks e.g., [[link]]
        ["w"] = {
            add = { "[[", "]]" },
            find = function()
                return require("nvim-surround.config").get_selection { motion = "2a]" }
            end,
            delete = "^(%[%[)().-(%]%])()$",
            change = {
                target = "^(%[%[)().-(%]%])()$",
                replacement = function()
                    return { { "[[" }, { "]]" } }
                end
            }

        }
    }
})
```
I can change other separators *into* wikilinks, but I can't change *from* wiklinks to other separators; this make me think that there is something wrong with `change.target`, but I have copied the expression from `delete`, so I think it should work. Do you know what may be the problem?

**Reply by kylechui** [reply]
Try omitting the `change` key altogether and see if that works.

**Reply by videbar** [reply]
Yep, that works. I guess that if `change` is not specified, it defaults to using the value of `add` for `replacement` and the value of `delete` for `target`?

**Reply by kylechui** [reply]
Yep, exactly
